jax/core.py

```diff
@@ -1633,10 +1633,6 @@ def symbolic_equal_shape(s1: Shape, s2: Shape) -> bool:
           all(unsafe_map(symbolic_equal_dim, s1, s2)))
 
 def greater_equal_dim(d1: DimSize, d2: DimSize) -> bool:
-  # TODO(mattjj): revise this temporary workaround for dynamic shapes
```

Comment:
I could not find a test that requires this. I believe that the right place to make this change is in partial_eval.DimensionHandlerTracer.greater_equal, and then the returned value should be False.